diesel/src/connection/sqlite/sqlite_value.rs

```diff
@@ -0,0 +1,92 @@
+extern crate libsqlite3_sys as ffi;
+extern crate libc;
+
+use std::ffi::CStr;
+use std::{slice, str};
+
+use backend::Sqlite;
+use row::Row;
+
+pub struct SqliteValue {
+    inner_statement: *mut ffi::sqlite3_stmt,
+    col_index: libc::c_int,
+}
+
+pub struct SqliteRow {
+    value: SqliteValue,
+    next_col_index: libc::c_int,
+}
+
+impl SqliteValue {
+    pub fn new(inner_statement: *mut ffi::sqlite3_stmt) -> Self {
+        SqliteValue {
+            inner_statement: inner_statement,
+            col_index: 0,
+        }
+    }
+
+    pub fn read_text(&self) -> Result<&str, str::Utf8Error> {
+        unsafe {
+            let ptr = ffi::sqlite3_column_text(self.inner_statement, self.col_index);
+            CStr::from_ptr(ptr as *const libc::c_char).to_str()
+        }
+    }
+
+    pub fn read_blob(&self) -> &[u8] {
+        unsafe {
+            let ptr = ffi::sqlite3_column_blob(self.inner_statement, self.col_index);
+            let len = ffi::sqlite3_column_bytes(self.inner_statement, self.col_index);
+            slice::from_raw_parts(ptr as *const u8, len as usize)
```

Comment:
Do you think that any docs/comments we'd add would clarify more than the `slice::from_raw_parts` documentation?